proxy/http/HttpTunnel.cc

```diff
@@ -1336,16 +1336,7 @@ HttpTunnel::consumer_handler(int event, HttpTunnelConsumer *c)
         c->producer->read_success = true;
         // Go ahead and clean up the producer side
         if (p->alive) {
-          p->alive = false;
-          if (p->read_vio) {
-            p->bytes_read = p->read_vio->ndone;
-          } else {
-            p->bytes_read = 0;
-          }
-          if (p->vc != HTTP_TUNNEL_STATIC_PRODUCER) {
-            // Clear any outstanding reads
-            p->vc->do_io_read(nullptr, 0, nullptr);
-          }
+          producer_handler(VC_EVENT_READ_COMPLETE, p);
```

Comment:
At this point, we may still have still have data left to read from the producer, right? How can we send an VC_EVENT_READ_COMPLETE to the producer?

Reply:
If the consumer is write_complete, it doesn't matter if there was more data on the producer, the consumer will never read it.  The tunnel will be stuck forever.

Reply:
I agree that consumer never reads it, but NetVC/CacheVC may still try to read the outstanding data and signal the continuation that VIO is holding and may result in crash if that continuation is already cleared.

Reply:
We should not leave around any vio's pointing to the continuation after we have freed the continuation.

Reply:
We have been running this commit in our 9.0.x since the end of 2020 without issue.